langwatch/src/server/tracer/tracesMapping.ts

```diff
@@ -405,6 +405,21 @@ export const TRACE_EXPANSIONS = {
   }
 >;
 
+/**
+ * Thread mappings for grouping traces by thread_id
+ * Single Responsibility: Define available mapping options for thread data structure
+ */
+export const THREAD_MAPPINGS = {
+  thread_id: {
+    mapping: (thread: { thread_id: string; traces: TraceWithAnnotations[] }) =>
+      thread.thread_id,
+  },
+  traces: {
+    mapping: (thread: { thread_id: string; traces: TraceWithAnnotations[] }) =>
+      thread.traces,
```

Comment:
to my previous comment, all that logic mapping the traces and so on should actually go here, no?